euphoria-operator-testkit/src/main/java/cz/seznam/euphoria/operator/test/accumulators/SingleJvmAccumulatorProvider.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.operator.test.accumulators;
+
+import cz.seznam.euphoria.core.client.accumulators.Accumulator;
+import cz.seznam.euphoria.core.client.accumulators.AccumulatorProvider;
+import cz.seznam.euphoria.core.util.Settings;
+
+import java.io.ObjectStreamException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * An accumulator provider gathering accumulators in-memory.<p>
+ *
+ * Safe for use only for unit testing purposes.
+ */
+public class SingleJvmAccumulatorProvider implements AccumulatorProvider {
```

Comment:
I would like to point out this `AccumulatorProvider` may be native for `in-mem executor`. So actually it may target wider usage possibilities than just unit testing.